src/main.rs

```diff
@@ -74,15 +93,17 @@ const RETRY_COUNT: usize = 5;
 async fn main() -> anyhow::Result<()> {
     setup_logger()?;
     let cli = Cli::parse();
-    if let Err(e) = run(cli).await {
-        log::error!("Error: {:?}", e);
-        std::process::exit(exitcode::SOFTWARE);
+    match run(cli).await {
```

Comment:
Maybe I'm missing something but if we already have result which can be either success or failure, why do we also need to return exit code? What would `Ok(EXIT_FAILURE)` mean? Does it mean we successfully captured failure of a test run and we want to propagate it to the caller of cli?

Reply:
We need to pipe the original exit code up so that we surface the expected results to the caller. We don't want to hide it if the results weren't quarantined.